packages/adblocker/src/lists.ts

```diff
@@ -189,18 +202,55 @@ export function parseFilters(
       const filter = NetworkFilter.parse(line, config.debug);
       if (filter !== null) {
         networkFilters.push(filter);
+        if (preprocessorStack.length) {
+          filterStack.push(filter);
+        }
       }
     } else if (filterType === FilterType.COSMETIC && config.loadCosmeticFilters === true) {
       const filter = CosmeticFilter.parse(line, config.debug);
       if (filter !== null) {
         if (config.loadGenericCosmeticsFilters === true || filter.isGenericHide() === false) {
           cosmeticFilters.push(filter);
+          if (preprocessorStack.length) {
+            filterStack.push(filter);
+          }
+        }
+      }
+    } else if (config.loadPreprocessors) {
```

Comment:
* This is a copy of Slack message

```
Ah, and about the config.loadPreprocessors . The intention of this flag is to remove side-effect of preprocessor processing in test/engine/bucket/filters.test.ts . I and 
@ chrmod
 validated that calling IFilter.getId in parseFilters will break the testing code because of updated id property.
```